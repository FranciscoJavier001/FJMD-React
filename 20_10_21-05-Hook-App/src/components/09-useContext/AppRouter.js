//** Routes lo unico que cambia es lo que es independiente ha esa pantalla, pero no va a hacer peticiones adicionales al servidor a menos que sea necesario, que no haga refresh del navegador web para cambiar entre pantallas */
//** Yo pongo div cuando tengo varios componentes */
//** High Order Component significa que tiene un componente adentro */

import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
  } from 'react-router-dom';

import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';

export const AppRouter = () => { //** Exportamos AppRouter a MainApp.js */
    return (
        <Router>
            <div>
                
                <NavBar />

                <div className="container">
                    {/* El Switch va a renderizar el componente selecciondo del path, por eso el navBar aparece arriba */}
                    <Switch>
                        {/* Ruta a la cual va a entrar y el que va a renderizar se pone en el component*/}
                        <Route exact path="/" component={ HomeScreen } />
                        
                        <Route exact path="/about" component={ AboutScreen } />
                        <Route exact path="/login" component={ LoginScreen } />
                        
                        <Redirect to="/" />

                    </Switch>
                </div>
            </div>
        </Router>
    )
}
