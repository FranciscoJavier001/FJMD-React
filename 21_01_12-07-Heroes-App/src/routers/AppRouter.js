//**_______________________________________________________________________________________________________________________________________________*/
//** Este sistema de rutas principal solo es que este tiene el router, es la unica diferencia */
import React, { useContext } from 'react' //** useContext nos permite acceder a datos globales desde cualquier componente */
import {
    BrowserRouter as Router, //** Le cambiamos el nombre a Router */
    Switch
  } from "react-router-dom";

  import { AuthContext } from '../auth/AuthContext'; //** Ahi creamos el createContext */
  import { PrivateRoute } from './PrivateRoute';
  import { PublicRoute } from './PublicRoute';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => { //** exporto este FC a HeroesApp */

    const { user } = useContext( AuthContext ) //** Accedemos al user que viene de src/auth/AuthContext, donde se creo "createContext" */
    // console.log(user); //** Asi vemos si el logged esta en false/true */

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute //** La Definimos en src/>routers/PublicRoute, esta pagina va a ser la primera */
                    exact //** Para que la Ruta sea Exacta */
                    path="/login" //** Cual va a ser la direccion */
                    component={ LoginScreen } //** Al componente donde voy a renderizar */
                    isAutenticated={ user.logged } //** Asi confirmamos si no esta autentificado isAutenticated, estado actual del usuario */
                    />

                    <PrivateRoute //** La Definimos en src/>routers/PrivateRoute, se accede a esta pagina al logearse, auqui hay mas componentes */
                    path="/" //** Esta es la ruta padre y pueda distribuirla a los hijos */
                    component={ DashboardRoutes } //** Aqui se Redirige el componente, aqui hay muchas rutas y se renderiza */
                    isAutenticated={ user.logged } //** Asi confirmamos si esta autentificado isAutenticated, porque ya tenemos context l18 */
                    />
                </Switch>
            </div>
    </Router>
    )
}