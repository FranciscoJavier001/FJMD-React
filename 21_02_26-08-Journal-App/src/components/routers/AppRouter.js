//** Atencion, mira estas son las rutas del AppRouter, como lo vemos no son tan complicadas, solo se necesita entender los parametros principales de hacia donde va la aplicacion */

import React, { useContext } from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom"

import { AuthRouter } from './AuthRouter'
import { JournalScreen } from '../journal/JournalScreen'

export const AppRouter = () => {
    return (
        <Router>
            <Switch>
                <Route
                path="/auth"
                component={ AuthRouter}
                />

                <Route
                exact
                path="/"
                component={ JournalScreen }
                />
                
                {/* Esto es para por si no encuentra ninguna ruta que nosotros definimos, en automatico nos mande aqui */}
                <Redirect to="/auth/login" />

            </Switch>
        </Router>
    )
}