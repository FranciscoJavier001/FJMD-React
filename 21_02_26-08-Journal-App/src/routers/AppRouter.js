//** Atencion, mira estas son las rutas del AppRouter, como lo vemos no son tan complicadas, solo se necesita entender los parametros principales de hacia donde va la aplicacion */

import React, { useContext, useEffect } from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom"

import { firebase } from '../firebase/firebase-config'

import { AuthRouter } from './AuthRouter'
import { JournalScreen } from '../components/journal/JournalScreen'

export const AppRouter = () => {

    //** Este efecto lo vamos a utilizar, para que aunque el usuario salga de la pagina, su sesion se mantenga */
    useEffect(() => { //** Por el momento aqui no voy a tener ninguna dependencia */
        //** Del archio de firebase hay un archivo importado de auth que lo vamos a utilizar */
        firebase.auth().onAuthStateChanged( (user) => {

            console.log(user);
        })
    }, [])

    return (
        <Router>
            <Switch>
                <Route
                path="/auth"
                component={ AuthRouter}
                />

                <Route
                exact
                path="/"
                component={ JournalScreen }
                />
                
                {/* Esto es para por si no encuentra ninguna ruta que nosotros definimos, en automatico nos mande aqui */}
                <Redirect to="/auth/login" />

            </Switch>
        </Router>
    )
}