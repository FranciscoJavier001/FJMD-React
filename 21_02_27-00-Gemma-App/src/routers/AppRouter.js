//** Este sistema de rutas principal solo es que este tiene el router, es la unica diferencia */
import React, { useContext } from 'react'
import {
    BrowserRouter as Router,
    Switch
  } from "react-router-dom";

  import { AuthContext } from '../auth/AuthContext';
  import { PrivateRoute } from './PrivateRoute';
  import { PublicRoute } from './PublicRoute';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => { //** Esto es un FuntionalComponent comun y corriente */

    // Para leer si esta autentificado
    const { user } = useContext( AuthContext ) //** Primero voy a utilizar el user y en el segundo va a ser el AuthContext */

    return (
        <Router>
            <div>
                <Switch>
                    {/* Aqui vamos a renderizar un componente, las rutas hijas */}
                    {/* El loginScreen no tiene ningun Navbar porque no tiene ningun estilo */}
                    <PublicRoute 
                    exact 
                    path="/login" 
                    component={ LoginScreen }
                    isAutenticated={ user.logged }
                    />
                    {/* Esta es una ruta que no esta definida, vamos a renderizar el DashboardRoutes cuando estamos en el path="/", ahora aqui ya vemos el NavBar */}
                    {/* Este componente lo vamos a validar, para que el usuario este loggeado para ingresar a la app, y a continuacion vamos a proteger la ruta, entonces vamos a renderizar el componente de manera condicional */}

                    {/* Esta la voy a proteger para proteger toda la App, cambio el Route que estaba al principio por el PrivateRoute */}
                    <PrivateRoute 
                    path="/" 
                    component={ DashboardRoutes }
                    //** Asi lo vamos a proteger con el isAutenticated */
                    isAutenticated={ user.logged }
                    />
                </Switch>
            </div>
    </Router>
    )
}