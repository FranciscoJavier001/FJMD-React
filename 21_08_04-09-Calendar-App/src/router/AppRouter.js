import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from 'react-router-dom';

import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch>
                    <Route exact path="/login" component={ LoginScreen } /> {/* Aqui voy a poner una ruta que va a ser exacta, con la direccion y el componente que vquiero mostrar */}
                    <Route exact path="/" component={ CalendarScreen } />

                    <Redirect to="/" />{/* Este lo uso para redireccionar al / */}
                </Switch>
            </div>
        </Router>
    )
}