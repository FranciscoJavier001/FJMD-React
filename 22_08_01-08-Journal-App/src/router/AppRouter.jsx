import React from 'react'

export const AppRouter = () => {
  return (
    <Routes>
        {/* Login y Registro */}
        <Route />

        {/* JournalApp */}
        <Route /> 
    </Routes>
  )
}
