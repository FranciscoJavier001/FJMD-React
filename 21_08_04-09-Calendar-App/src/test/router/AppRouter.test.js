//**_______________________________________________________________________________________________________________________________________________*/
import Enzyme from 'enzyme'; //** Si no tengo estos me manda falla */
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'; //** Si no tengo estos me manda falla */

Enzyme.configure({ adapter: new Adapter() }); //** Si no tengo estos me manda falla */

import React from 'react' //** Uso React porque voy a renderizar un componente */
import { mount } from 'enzyme'
import { Provider } from 'react-redux' //** Para utilizar el mount necesito un componente llamado Provider */

import configureStore from 'redux-mock-store' //** Para configurar el store */
import thunk from 'redux-thunk' //** La de arriba ocupa esta porque las acciones retornan una funcion */

import '@testing-library/jest-dom' //** Ayuda con el tipado */
import { AppRouter } from '../../router/AppRouter';

const middlewares = [ thunk ] //** Funcion que se invoca despues de que se envia una accion, puede modificarla, esperar que termine o cancelarla */
const mockStore = configureStore( middlewares ) //** MockStore es un objero que simula ser otro y el store configura funciones de los moddleware */

// store.dispatch = jest.fn() //** Puedo saber con que argumento se utilizo para llamar el dispatch, por eso lo manejo con un mock */


describe('Pruebas en <AppRouter />', () => {
    
    test('debe de mostrar el espere...', () => {

        const initState = { //** Configuro el iS, que asi quiero cuando inicie las pruebas */
            auth: { //** Falla, no puede desestructirar el checking */
                checking: true
            }
        }
        
        const store = mockStore( initState ) //** Este va con el berofeEach para inicializar todas las acciones que este store haya ejecutado */
        
        const wrapper = mount( //** Dentro necesito el componente llamado Provider */
            <Provider store={ store }> {/* Esto me va a proveer el store, pero debo de crearlo, dentro mando el store creado anteriormente */}
                <AppRouter /> {/* Dentro de este HOC voy a colocar el evento a analizar */}
            </Provider>
        )

        // expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('h5').exists() ).toBe(true) //** Busca un h5 y si existe pasa la prueba */
    })

    test('debe de mostrar la ruta publica', () => {

        const initState = { //** Configuro el iS, que asi quiero cuando inicie las pruebas */
            auth: { //** Falla, no puede desestructirar el checking */
                checking: false, //** Lo pongo en false, porque ya paso el espere */
                uid: null //** Porque el usuario no esta autentificado */
            }
        }

        const store = mockStore( initState ) //** Este va con el berofeEach para inicializar todas las acciones que este store haya ejecutado */
        
        const wrapper = mount( //** Dentro necesito el componente llamado Provider */
            <Provider store={ store }> {/* Esto me va a proveer el store, pero debo de crearlo, dentro mando el store creado anteriormente */}
                <AppRouter /> {/* Dentro de este HOC voy a colocar el evento a analizar */}
            </Provider>
        )

        expect( wrapper ).toMatchSnapshot() //** Voy a esperar que haga match con el snapshot */
        expect( wrapper.find('.login-container').exists() ).toBe(true) //** Busca un login-container(LoginScreen) y si existe pasa la prueba */
    })

    test('debe de mostrar la ruta privada', () => {

        const initState = { //** Configuro el iS, que asi quiero cuando inicie las pruebas */
            calendar: { //** No podia desestructurar las propiedades de los events, asi que lo proveo aqui como un arreglo vacio*/
                events: []
            },
            ui: {//** Proveer el modalOpen, lo pongo en falso, porque aqui ya entro */
                modalOpen: false 
            },
            auth: { //** Falla, no puede desestructirar el checking */
                checking: false, //** Lo pongo en false, porque ya paso el espere */
                uid: '123', //** Existe porque el usuario esta autentificado */
                name: 'Juan Carlos' //** Porque debe de venir el nombre */
            }
        }

        const store = mockStore( initState ) //** Este va con el berofeEach para inicializar todas las acciones que este store haya ejecutado */
        
        const wrapper = mount( //** Dentro necesito el componente llamado Provider */
            <Provider store={ store }> {/* Esto me va a proveer el store, pero debo de crearlo, dentro mando el store creado anteriormente */}
                <AppRouter /> {/* Dentro de este HOC voy a colocar el evento a analizar */}
            </Provider>
        )

        expect( wrapper ).toMatchSnapshot() //** Voy a esperar que haga match con el snapshot */
        expect( wrapper.find('.calendar-screen').exists() ).toBe(true) //** Busca un calendar-screen(CalendarScreen) y si existe pasa la prueba */
    })
    
})
