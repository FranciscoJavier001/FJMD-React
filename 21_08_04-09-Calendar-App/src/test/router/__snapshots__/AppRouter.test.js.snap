// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar la ruta privada 1`] = `ReactWrapper {}`;

exports[`Pruebas en <AppRouter /> debe de mostrar la ruta publica 1`] = `ReactWrapper {}`;
